components/homme/src/share/cxx/Hommexx_config.h.in

```diff
@@ -1,9 +1,6 @@
 #ifndef HOMMEXX_CONFIG_H
 #define HOMMEXX_CONFIG_H
 
-// Identifies whether this is a cuda/hip build
-#cmakedefine HOMMEXX_ENABLE_GPU
```

Comment:
Is this removal intended? There seems to be plenty of code that depends on this macro being defined/undefined.

Reply:
This symbol was being defined in both Hommexx_config.h and config.h. The distinction between these two is as follows, as I understand things. The first is for symbols used only in the Hommexx code. The second is for symbols used in both Hommexx and Homme, the latter including Homme dycores built for BFB testing with Hommexx.

Since the symbol is redundantly defined (leading to a preprocessor warning about this), one should be removed. The symbol is used in Homme as well as Hommexx, so I believe the Hommexx_config.h.in instance should be removed.

My SCREAM test on Ascent, still to do, will check this point.

Reply:
Ok, I think I see why this symbol got redundantly defined. The SCREAM build indeed is not seeing the symbol. I'm going to investigate to see if I can do something about that. In the worst case I'll revert this commit, since redundant defines are not pretty but also not fatal.

Reply:
I think I have a way to do this to avoid redefining this symbol. Will test tomorrow.

Reply:
The last commit has resolved this both issues: working in SCREAM and avoiding redundant defines in standalone HOMMEXX.